### PR TITLE
DCS-1379 extracted temporary absences package and added getTemporaryAbsence endpoint

### PR DESCRIPTION
**Pull request by louisenorris**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/resource/TemporaryAbsencesResourceTest.kt [hunk 1]
```diff
@@ -53,7 +53,39 @@ class TemporaryAbsencesResourceTest : IntegrationTestBase() {
...
+          prisonNumber = "A1234AA",
+          reasonForAbsence = "Hospital"
+        )
+      )
+      webTestClient.get().uri("/temporary-absences/MDI/A1234AA")
+        .headers(setAuthorisation(roles = listOf("ROLE_VIEW_ARRIVALS"), scopes = listOf("read")))
+        .exchange()
+        .expectStatus().isOk
+        .expectBody().json("temporaryAbsence".loadJson(this))
+    }
```
Comment (andrewrlee) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/resource/TemporaryAbsencesResourceTest.kt:79]: Maybe best to add tests for?:
* `requires correct role`
* `requires authentication`

File: src/test/resources/uk/gov/justice/digital/hmpps/welcometoprison/resource/temporaryAbsence.json [hunk 2]
```diff
@@ -1,16 +1,7 @@
...
-    "reasonForAbsence": "Dentist"
-  }
-]
+{
+  "firstName": "Jim",
+  "lastName": "Smith",
+  "dateOfBirth": "1991-07-31",
+  "prisonNumber": "A1234AA",
+  "reasonForAbsence": "Hospital"
+}
```
Comment (andrewrlee) [inline src/test/resources/uk/gov/justice/digital/hmpps/welcometoprison/resource/temporaryAbsence.json:7]: meh.. ⛔

**Review by andrewrlee — APPROVED** [review]
LGTM

File: src/test/resources/uk/gov/justice/digital/hmpps/welcometoprison/resource/temporaryAbsence.json (hunk 2 above)
Comment (louisenorris) [reply src/test/resources/uk/gov/justice/digital/hmpps/welcometoprison/resource/temporaryAbsence.json:7]: Arg - sorry

File: src/test/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/resource/TemporaryAbsencesResourceTest.kt (hunk 1 above)
Comment (louisenorris) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/resource/TemporaryAbsencesResourceTest.kt:111]: Added in latest commit

File: src/test/resources/uk/gov/justice/digital/hmpps/welcometoprison/resource/temporaryAbsence.json (hunk 2 above)
Comment (louisenorris) [reply src/test/resources/uk/gov/justice/digital/hmpps/welcometoprison/resource/temporaryAbsence.json]: Fixed in latest commit

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**